packages/components/image/src/image.vue

```diff
@@ -0,0 +1,265 @@
+<template>
+  <div
+    ref="container"
+    :class="['el-image', $attrs.class]"
+    :style="containerStyle"
+  >
+    <slot v-if="loading" name="placeholder">
+      <div class="el-image__placeholder"></div>
+    </slot>
+    <slot v-else-if="hasLoadError" name="error">
+      <div class="el-image__error">{{ t('el.image.error') }}</div>
+    </slot>
+    <img
+      v-else
+      class="el-image__inner"
+      v-bind="attrs"
+      :src="src"
+      :style="imageStyle"
+      :class="{
+        'el-image__preview': preview,
+      }"
+      @click="clickHandler"
+    />
+    <teleport to="body" :disabled="!appendToBody">
+      <template v-if="preview">
+        <image-viewer
+          v-if="showViewer"
+          :z-index="zIndex"
+          :initial-index="imageIndex"
+          :url-list="previewSrcList"
+          :hide-on-click-modal="hideOnClickModal"
+          @close="closeViewer"
+          @switch="switchViewer"
+        >
+          <div v-if="$slots.viewer">
+            <slot name="viewer" />
+          </div>
+        </image-viewer>
+      </template>
+    </teleport>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ref, onMounted, watch, nextTick } from 'vue'
+import { isString } from '@vue/shared'
+import { useEventListener, useThrottleFn } from '@vueuse/core'
+import { useAttrs, useLocaleInject } from '@element-plus/hooks'
+import ImageViewer from '@element-plus/components/image-viewer'
+import isServer from '@element-plus/utils/isServer'
+import { getScrollContainer, isInContainer } from '@element-plus/utils/dom'
+import { imageEmits, imageProps } from './image'
+
+import type { CSSProperties, StyleValue } from 'vue'
+
+const isHtmlElement = (e: any): e is Element =>
+  e && e.nodeType === Node.ELEMENT_NODE
+
+let prevOverflow = ''
+
+export default defineComponent({
+  name: 'ElImage',
+  components: {
+    ImageViewer,
+  },
+  inheritAttrs: false,
+
+  props: imageProps,
+  emits: imageEmits,
+
+  setup(props, { emit, attrs: rawAttrs }) {
+    const { t } = useLocaleInject()
+
+    const attrs = useAttrs()
+    const hasLoadError = ref(false)
+    const loading = ref(true)
+    const imgWidth = ref(0)
+    const imgHeight = ref(0)
+    const showViewer = ref(false)
+    const container = ref<HTMLElement>()
+
+    const _scrollContainer = ref<HTMLElement | Window>()
+    let stopScrollListener: () => void
+    let stopWheelListener: () => void
+
+    const containerStyle = computed<StyleValue>(
```

Comment:
I think the annotation after `computed` seems redundant here since you have already used `as` operator to cast the return type.

Reply:
Done.